### ISSUE TITLE
Is a Mac version possible?

### ISSUE THREAD

**Issue by Dunkirk** [LoneGazebo/Community-Patch-DLL#4686]
With the recent push by Microsoft to make .NET Core and Visual Studio Code cross-platform, is it possible to build a Mac version of the library which would make the mod work on macOS? I've done a little work in compiling originally-Windows-based C# programs on Mac with Visual Studio, and it's my understanding that it is possible to create library targets from C++ code with Visual Studio Core. However, that still doesn't actually answer the question. Is there anything in the code that requires .NET Standard components, which will not be available in .NET Core? And, even if the code will compile into a shared object, could the mod use a Mac version of the library, or does it require a DLL, no matter what?

**Comment by supermerill** [comment]
The mac version is made by Aspire. It's statically linked. Can't make a shared library for mac, as it's not used. .net is not used by civ5.

You can install steam on wine and then civ5 on this steam, then VP on wine. Should work.
You can also wait and hope that proton (wine for steam) reach a state where it's possible to install the windows version on a mac.

**Comment by Dunkirk** [comment]
However the Mac version of Civ 5 is programmed or compiled, it can use mods. Are you trying to tell me that the Mac version can't reference ANY shared library in ANY of its mods? That may be true, and that's the first question.

I'm a little fuzzy here, but it should only matter if VoxPopuli.vcxproj can be compiled on a Mac. At the end of the day, .NET is irrelevant. It's just C++ code. The only thing that would prevent it from compiling on Mac (or Linux) would be Windows-specific function calls. But if the mod never actually calls Windows system functions -- if it's only self-referencing (and talking to the main loop in Civ 5) -- then I would expect it to compile.

In other words: Is the DLL in this mod "just" a plain old C++ program? If so, it should compile. If it compiles, it can be made a shared object, which the Mac version of a .dll. If it can be made a .so, can the Firaxis mod system on a Mac use it, or is that system somehow limited to only calling Windows-specific DLL's? I'm totally ignorant of how the modding system works here.

If the mod library code compiles, and the Mac version of Civ 5 can reference a .so, then would this not be possible to package for a Mac?

(There's also the whole question of how Lua fits into all of this, but Lua runs on everything.)

**Comment by ilteroi** [comment]
we need to use the vc++ 2008 toolset for compiling and linking against some other parts of the SDK for which the source code wasn't provided. i don't see a way to make it work natively on mac. but wine is quite good these days, so why not use that?

**Comment by Dunkirk** [comment]
There it is. That would be the kind of show-stopper I was looking for, before I tried going down the rabbit hole. Thanks!
